### PR TITLE
feat(recall): MZI-187 fetch recalled mails from db

### PR DESCRIPTION
**Pull request by aymericdevriesere**
## Describe your changes
Add all classes and logic needed to request database and retrieve recallTasks.
This PR is part of the update how do we manage recall mail.
## Issue ticket number and link
[ MZI-187 ]
https://jira.support-ent.fr/browse/MZI-187
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import org.entcore.common.sql.SqlResult;
+
+public class SqlRecallTaskService extends DbTaskService<RecallTask> {
+    private final String recallMailTable = schema + "." + "recall_mails";
+    public SqlRecallTaskService(String schema) {
+        super(schema);
+    }
+
+    @Override
+    public Future<JsonArray> retrieveTasksDataFromDB(TaskStatus status) {
```
Comment (BastienBenard) [inline zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java:23]: Tu peux utiliser des model en retournant une Future\<List\<RecallTask\>\>

File: zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                .append("WHERE recall_tasks.status = ?;");
+        JsonArray params = new JsonArray().add(status.method());
+
+        Sql.getInstance().prepared(query.toString(), params, SqlResult.validResultHandler(PromiseHelper.handlerJsonArray(promise)));
+
+        return promise.future();
+    }
+
+    @Override
+    public Future<JsonObject> createTask(Action<RecallTask> action, RecallTask task) {
```
Comment (BastienBenard) [inline zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java:39]: Tu peux utiliser des model en retournant une Future\<<RecallTask\>

File: zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public Future<JsonArray> retrieveTasksDataFromDB(TaskStatus status) {
+        Promise<JsonArray> promise = Promise.promise();
+        StringBuilder query = new StringBuilder();
+
+        query.append("SELECT * ")
+                .append(this.taskTable + " as recall_tasks ")
+                .append("JOIN " + this.actionTable + " as actions" + " on actions.id = recall_tasks.action_id")
+                .append("JOIN " + this.recallMailTable + " as recall_mails" + " on actions.id = recall_mails.action_id ")
+                .append("WHERE recall_tasks.status = ?;");
```
Comment (JinDouang) [inline zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java]: Avec StringBuilder, qu'est-ce que tu utilises de plus hormis `.append` ? 
Si tu n'as pas d'autres usage (`insert`, `reverse` etc...) selon moi il vaut mieux utiliser un String normal : 

```suggestion
 String query = "SELECT * " +
                this.taskTable + " as recall_tasks " +
                "JOIN " + this.actionTable + " as actions" + " on actions.id = recall_tasks.action_id" +
                "JOIN " + this.recallMailTable + " as recall_mails" + " on actions.id = recall_mails.action_id " +
                "WHERE recall_tasks.status = ?;";
```

File: zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        return promise.future();
+    }
+
+    @Override
+    public Future<JsonObject> createTask(Action<RecallTask> action, RecallTask task) {
+        Promise<JsonObject> promise = Promise.promise();
+
+        StringBuilder query = new StringBuilder();
+
+        query.append("INSERT INTO ")
```
Comment (JinDouang) [inline zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java]: cf commentaire

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Review by BastienBenard — APPROVED** [review]

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**